package/com.unity.formats.usd/Tests/Editor/USDPayloadComponentTests.cs

```diff
@@ -210,35 +210,92 @@ public IEnumerator IsLoaded_IsFalse_WhenIsUnloadedFromComponent()
         [UnityTest]
         public IEnumerator ChangingUsdPayloadStateFromUnloadedToLoaded_MarksSceneDirty()
         {
+            const int timeOutSeconds = 3;
+
             m_usdAsset.m_payloadPolicy = PayloadPolicy.DontLoadPayloads;
             m_usdAsset.Reload(forceRebuild: true);
             yield return null;
 
             EditorSceneManager.SaveScene(m_UnityScene, GetUnityScenePath());
             Assert.IsFalse(m_UnityScene.isDirty, "Scene should not be dirty after after saving");
 
-            // Change payload state and wait one frame to ensure Update runs
+            float loadingTime = 0;
             m_usdRoot.GetComponentInChildren<UsdPayload>().Load();
+
+            while (!m_UnityScene.isDirty && loadingTime < timeOutSeconds)
+            {
+                loadingTime += Time.deltaTime;
+                yield return null;
+            }
+
+            Assert.Less(loadingTime, timeOutSeconds);
+            Assert.True(m_UnityScene.isDirty);
+        }
+
+        [UnityTest]
+        public IEnumerator ChangingUsdPayloadStateFromUnloadedToLoaded_GeneratesPayloadObjects()
+        {
+            m_usdAsset.m_payloadPolicy = PayloadPolicy.DontLoadPayloads;
+            m_usdAsset.Reload(forceRebuild: true);
             yield return null;
 
-            Assert.IsTrue(m_UnityScene.isDirty, "Scene should be dirty after changing UsdPayload objects");
+            EditorSceneManager.SaveScene(m_UnityScene, GetUnityScenePath());
+            Assert.IsFalse(m_UnityScene.isDirty, "Scene should not be dirty after after saving");
```

Comment:
I know this was already there, but as you're in this code could you remove the extra "after" here please? :)